### PR TITLE
Fixes #12604: Generation gets stuck when cf-serverd is not running

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/12604

There was a bug in NuProcess that was making child process inherit all file descriptors of the parent. They have corrected it 2 weeks ago :)

### CODE REVIEW

File: rudder-core/pom.xml [hunk 1]
```diff
@@ -159,33 +159,33 @@ along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
...
       <version>2.3.5</version>
     </dependency>
-    
+
     <!-- High performance, low memory native process fork/exec -->
     <dependency>
         <groupId>com.zaxxer</groupId>
         <artifactId>nuprocess</artifactId>
-        <version>1.1.0</version>
+        <version>1.2.0</version>
```
Comment (fanf) [inline rudder-core/pom.xml]: Yep. The whole correction.

**Comment by fanf** [comment]
PR rebased

File: rudder-core/src/test/scala/com/normation/rudder/hooks/RunNuCommandTest.scala [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    "has only the environment variable explicitly defined" in {
+      RunNuCommand.run(Cmd("/usr/bin/env", Nil, Map("foo" -> "bar"))) must beEqualTo(CmdResult(0, "foo=bar\n", "")).awaitFor(100.millis)
+    }
+
+    "has only the 3 stdin, stdout, stderr file descriptors" in {
+      //we only keep the return code and the number of lines in the ls output
+      //(we have one more line because last line ends with a "\n"
+      val res = RunNuCommand.run(Cmd("/usr/bin/ls", "-1" :: "/proc/self/fd" :: Nil, Map())).map { cmd =>
```
Comment (ncharles) [inline rudder-core/src/test/scala/com/normation/rudder/hooks/RunNuCommandTest.scala]: this is not distribution agnostic: on Debian it's /bin/ls
Can it use PATH ?

**Comment by fanf** [comment]
PR rebased

**Comment by fanf** [comment]
OK, merging this PR

**Pull request merged**